lang/py3/build.sh

```diff
@@ -17,49 +17,23 @@
 
 set -e # exit on error
 
-function usage {
+usage() {
   echo "Usage: $0 {test|dist|clean}"
-  exit 1
 }
 
-if [ $# -eq 0 ]
-then
-  usage
-fi
-
-if [ -f VERSION.txt ]
-then
-  VERSION=`cat VERSION.txt`
-else
-  VERSION=`cat ../../share/VERSION.txt`
-fi
-
-for target in "$@"
-do
-
-function do_clean(){
-  python3 setup.py clean
-  rm -rvf dist avro_python3.egg-info avro/*.avsc avro/VERSION.txt avro/__pycache__/ avro/tests/interop.avsc avro/tests/__pycache__/
-}
-
-case "$target" in
-  test)
-    python3 setup.py test
-    ;;
-
-  dist)
-     python3 setup.py sdist
-     cp -r dist ../../dist/py3
-    ;;
-
-  clean)
-    do_clean
-    ;;
-
-  *)
+main() {
+  local target
+  if (( $# == 0 )); then
     usage
-esac
-
-done
+    return 1
+  fi
+  for target; do
+    case "$target" in
+      clean|dist|test) : ;;
+      *) usage; return 1 ;;
+    esac
+  done
```

Comment:
Yeah, it's a tradeoff I'm not entirely sure I like. In its current form it supports more than one argument, just like ./setup.py itself does.

```bash
./setup.py clean test
```
is valid, and so is
```bash
build.sh clean test
```
but if I were to do
```bash
while [[ $1 ]]; do
    case "$1" in ...; esac
    shift
done
```
or even do away with the loop/shift entirely and do
```bash
case "$1" in ...
```
that would not work. On the other hand, `build.sh` is not entirely transparent to `./setup.py`, because you can do
```bash
./setup.py clean -vvv
```
but you can't do
```bash
./build.sh clean -vvv
```

As a project-standard interface across different languages, I think ./build.sh has to support multiple arguments.

Reply:
Oh, I see, yeah. I didn't catch the intent. Thanks. 👍